src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/TypeBuilderImpl.cs

```diff
@@ -564,11 +710,184 @@ public override ConstructorInfo[] GetConstructors(BindingFlags bindingAttr)
+
+                if (targetMethod == null)
+                {
+                    Type? parent = _typeParent;
+                    while (parent != null && targetMethod == null)
+                    {
+                        targetMethod = parent.GetMethod(interfaceMethod.Name, GetBindingFlags(interfaceMethod), null, interfaceMethod.CallingConvention, GetParameterTypes(interfaceMethod.GetParameters()), null);
+                        parent = parent.BaseType;
+                    }
+                }
+
+                im.InterfaceMethods[i] = interfaceMethod;
+                im.TargetMethods[i] = targetMethod!;
+            }
+
+            if (_interfaceMappings == null)
+            {
+                return im;
+            }
+
+            // Add the interface implementations defined with 'DefineMethodOverride',
+            // this overrides default implementations if there is any
+            if (_interfaceMappings.TryGetValue(interfaceType, out var mapping))
+            {
+                for (int i = 0; i < interfaceMethods.Length; i++)
+                {
+                    for (int j = 0; j < mapping._interfaceMethods.Count; j++)
+                    {
+                        if (im.InterfaceMethods[i].Equals(mapping._interfaceMethods[j]))
+                        {
+                            im.TargetMethods[i] = mapping._targetMethods[j];
+                        }
+                    }
+                }
+            }
+
+            return im;
+        }
+
+        private static Type[] GetParameterTypes(ParameterInfo[] parameterInfos)
+        {
+            Type[] parameterTypes = new Type[parameterInfos.Length];
+            for (int i = 0; i < parameterInfos.Length; i++)
+            {
+                parameterTypes[i] = parameterInfos[i].ParameterType;
+            }
+            return parameterTypes;
+        }
+
+        private void ValidateInterfaceType(Type interfaceType)
+        {
+            if (!interfaceType.IsInterface)
+            {
+                throw new ArgumentException(SR.Argument_MustBeInterface, nameof(interfaceType));
+            }
+
+            if (!IsInterfaceImplemented(interfaceType))
+            {
+                throw new ArgumentException(SR.Argument_InterfaceNotFound, nameof(interfaceType));
+            }
+
+            if (IsGenericParameter)
+            {
+                throw new InvalidOperationException(SR.Argument_GenericParameter);
+            }
+
+            // SZArrays implement the methods on IList`1, IEnumerable`1, and ICollection`1 with
+            // SZArrayHelper and some runtime magic. We don't have accurate interface maps for them.
+            if (IsSZArray && interfaceType.IsGenericType)
+            {
+                throw new ArgumentException(SR.Argument_ArrayGetInterfaceMap);
```

Comment:
Can this throw with "normal" input?

Reply:
Normal would be ECMA compliant and not corrupt or generally invalid input.

Reply:
Is this reachable code - do the test exercise this ArgumentException? 

You are changing TypeBuilderImpl.IsSZArray to return false in this PR. If I am reading the code correctly, this should be unreachable and it can be deleted.

Reply:
> You are changing TypeBuilderImpl.IsSZArray to return false in this PR. If I am reading the code correctly, this should be unreachable and it can be deleted.

Right, this TypeBuilderImpl instance will never be an array, so unreachable can be deleted, this validation copied from RuntimeType. GetInterfaceMap(...) validation